kratos/utilities/math_utils.h

```diff
@@ -279,20 +279,14 @@ class MathUtils
     static inline BoundedMatrix<TDataType, TDim, TDim> InvertMatrix(
             const BoundedMatrix<TDataType, TDim, TDim>& InputMatrix,
             TDataType& InputMatrixDet,
-            const TDataType Tolerance = ZeroTolerance
+            const TDataType MaxCondNumber = (1.0/std::numeric_limits<double>::epsilon()) * 1.0e-4 // we want at least 4 significant digits.
```

Comment:
Use max instead https://en.cppreference.com/w/cpp/types/numeric_limits/max

Reply:
no, the max is 1e308, I need the highest value that can be compared against 1. The inverse of epsilon, around 1e16.